From david-b@pacbell.net  Thu Jul  2 15:45:24 2009
From: Qiuping Chen <cqiuping@gmail.com>
Date: Wed, 1 Jul 2009 03:49:29 -0700
Subject: USB: gadget: rndis conformance tweak
To: Greg KH <greg@kroah.com>
Cc: linux-usb@vger.kernel.org, Qiuping Chen <cqiuping@gmail.com>, Helen Chen <helen.chen@intel.com>
Message-ID: <200907010349.29608.david-b@pacbell.net>
Content-Disposition: inline


From: Qiuping Chen <cqiuping@gmail.com>

Support OID_802_3_MAC_OPTIONS in gen_ndis_query_resp() of rndis.c to
make RNDIS gadget pass 1c_SetMulticast subtest in Microsoft NDISTest6:

  http://www.microsoft.com/whdc/DevTools/tools/NDIStest.mspx

The other tests in NDISTest6 are passed.

[ dbrownell@users.sourceforge.net: remove OID_802_3_MAXIMUM_LIST_SIZE
setting ... it was bogus, this code only handles one entry, not 32.
And we don't know what would break if we lied about that... ]

Signed-off-by: Helen Chen <helen.chen@intel.com>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/rndis.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/gadget/rndis.c
+++ b/drivers/usb/gadget/rndis.c
@@ -442,6 +442,8 @@ gen_ndis_query_resp (int configNr, u32 O
 
 	case OID_802_3_MAC_OPTIONS:
 		pr_debug("%s: OID_802_3_MAC_OPTIONS\n", __func__);
+		*outbuf = cpu_to_le32(0);
+		retval = 0;
 		break;
 
 	/* ieee802.3 statistics OIDs (table 4-4) */
